src/rules/newlineBeforeReturnRule.ts

```diff
@@ -77,10 +84,18 @@ class NewlineBeforeReturnWalker extends Lint.AbstractWalker<void> {
             }
         }
         const prevLine = ts.getLineAndCharacterOfPosition(this.sourceFile, prev.end).line;
-
         if (prevLine >= line - 1) {
+            const indentationCurrent = this.getIndentation(node);
+            const indentationPrev = this.getIndentation(prev);
+
+            const fixer = Lint.Replacement.replaceFromTo(
+                start - indentationCurrent.length,
+                start,
+                line === prevLine ? `\n\n${indentationPrev}` : `\n${indentationCurrent}`,
```

Comment:
> `\n`

This'll need to adjust to projects where `\r\n` is the line feed, such as Git with CRLF (aka Windows). `curlyRule.ts` checks the last character of a relevant source line to see which to use, for example.